consensus/istanbul/backend/announce.go

```diff
@@ -125,14 +125,12 @@ func (sb *Backend) announceThread() {
 		case <-checkIfShouldAnnounceTicker.C:
 			logger.Trace("Checking if this node should announce it's enode")
 
-			shouldAnnounce, err = sb.shouldSaveAndPublishValEnodeURLs()
-			if err != nil {
-				logger.Warn("Error in checking if should announce", err)
-				break
-			}
+			shouldQuery = sb.shouldParticipateInAnnounceAnnounce()
```

Comment:
How come you need to split up these periodic query and periodic announce version update?

Reply:
Replicas need to query to populate their validator enode table (specifically the encrypted enode urls), but don't want to update the proxie's validator assignments at the same time as the primary. Also added some more comments in the code.